workspaces/util/src/__tests__/distinctArray-test.ts

```diff
@@ -0,0 +1,75 @@
+import { describe, expect, it } from "@jest/globals";
+import { distinctArray } from "../distinctArray";
+
+describe("distinctArray", () => {
+    it("returns an empty list if given an empty list", () => {
+        expect(distinctArray([])).toStrictEqual([]);
+    });
+
+    it("can handle NaN in an iterable of numbers", () => {
+        expect(distinctArray([156, 89, NaN, 99, NaN, 156])).toStrictEqual([156, 89, NaN, 99]);
+    });
+
+    it("can handle an iterable of booleans", () => {
+        expect(distinctArray([true, true, false, false])).toStrictEqual([true, false]);
+    });
+
+    it("can handle iterables like sets", () => {
+        expect(
+            distinctArray(
+                new Set(["uno", "dos", "tres", "quatro", "cinco", "cinco", "seis"])
+            )
+        ).toStrictEqual(["uno", "dos", "tres", "quatro", "cinco", "seis"]);
+    });
+
+    it("is not guaranteed to receive distinct result for a map with no keyFn", () => {
```

Comment:
So for a Map used like this, it's actually guaranteed to not de-duplicate anything. That's because a Map when treated as an iterable gives us the entries, and each entry is going to be an array, and no two arrays are equal to each other unless they're the same reference.

I think the test should be reframed to say something along the lines of "it defaults to comparing objects by reference", and then we could have a test with an array of objects where some of them are the same reference, and then it gets de-duplicated; whereas if we have a clone of some object but a different reference, they are both still included.